pkg/storj/node.go

```diff
@@ -3,13 +3,174 @@
 
 package storj
 
-import "encoding/hex"
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"math/bits"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/sha3"
+	"storj.io/storj/pkg/utils"
+)
+
+// IdentityLength is the number of bytes required to represent a node id
+const IdentityLength = 32
+
+var (
+	ErrNotImplemented = errs.Class("not implemented error")
+	// ErrNodeID is used when something goes wrong with a node id
+	ErrNodeID   = errs.Class("node ID error")
+	// EmptyNodeID is the zero-value for a NodeID
+	EmptyNodeID = NodeID([IdentityLength]byte{})
+)
 
 // NodeID is a unique node identifier
-type NodeID [32]byte
+type NodeID [IdentityLength]byte
+type NodeIDList []NodeID
```

Comment:
Where do we need the NodeIDList definition? Also thinking whether `NodeIDs` would be nicer, but `NodeIDList` is good as well.

Reply:
Hmm, can we use `sort.Slice(nodeids, func(i, k int) bool {  return nodeids[i].Less(nodeids[k]) }` there?

Reply:
it's used in a few places